data_fetcher.py

```diff
@@ -0,0 +1,110 @@
+import os
+import shutil
+import json
+import pandas as pd
+import yfinance as yf
+from datetime import datetime
+
+class DataFetcher:
+    """
+    This module downloads data from yahoo finance and saves it locally in a downloads directory
+    """
+    def __init__(self, directory_path: str, import_start_date: datetime=datetime(2008,1,1)) -> None:
```

Comment:
In my settings code i used datetime.date while you are using datetime.datetime. I don't this there is a use for the "time" part. Nevertheless we should decide which one to use.

Reply:
we'll merge your branch into main, then I will make adaptations